transcom/mymove
# B-17907-INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a867936)

## Summary

This work adds move history tracking for lots of events related to customer PPM closeout. Move history SQL was updated to include address changes to the W2 address and file uploads related to PPM submission. Front end templates were added for all the new events.

### How to test

If you have a move in the PPM closeout stage, please jump to step 4
1. Login as a customer and create a new move with a PPM shipment
2. Fill out the required details and submit
3. Login as a service counselor, fill out the required information and approve the move
4. Login as a customer with a PPM in the closeout stage and begin the closeout process
6. Fill out the required information regarding actual departure date, zip code, advances, w2 address, etc
7. Fill out the required information for a trip and make uploads for empty, full, and trailer weights
8. Create a pro-gear set for yourself, and upload a weight ticket
9. Create a pro-gear set for your spouse, and upload a weight ticket
10. Create an expense and upload a receipt
11. Make edits to all of the above items (address info, trips, pro gear sets, expenses) NOTE: you cannot delete an upload - this is a known issue and should not block this work
12. Make a second trip and upload weight tickets against it
13. Delete a trip, a pro-gear set, and an expense
14. Login as a QAE and locate the move
15. Navigate to move history and ensure all events are tracked:
- When a customer has Uploaded documents (weight tickets, receipts, pro-gear weights)
- When a customer has updated PPM details (departure date, locations, advance, W-2 address)
- When a customer has updated PPM details (Weight tickets, Expenses, Pro-gear)
- Actual Calculated Incentives and Reimbursements, after a customer has submitted the PPM
